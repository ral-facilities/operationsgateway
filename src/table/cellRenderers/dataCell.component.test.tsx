import React from 'react';
import DataCell, { DataCellProps } from './dataCell.component';
import { render, RenderResult } from '@testing-library/react';

describe('Data Cell', () => {
  let props: DataCellProps;

  const createView = (): RenderResult => {
    return render(
      <table>
        <tbody>
          <tr>
            <DataCell {...props} />
          </tr>
        </tbody>
      </table>
    );
  };

  beforeEach(() => {
    props = {
      dataKey: 'test',
      rowData: 'test data',
    };
  });

  it('renders correctly', () => {
    const view = createView();
    expect(view.asFragment()).toMatchSnapshot();
  });

  it.todo('more tests');
});
