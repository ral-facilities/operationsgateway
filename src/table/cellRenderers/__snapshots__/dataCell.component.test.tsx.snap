// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Cell renders correctly 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-100wy9x-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-z1doim-MuiTypography-root"
              >
                test data
              </p>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
